docs/api/repository-signatures-resource.md

```diff
@@ -0,0 +1,114 @@
+ms.reviewer:
+- karann
+- unniravindranathan
+---
+
+# Repository signatures
+
+If a package source supports adding repository signatures to published packages, it is possible for a client to
+determine the signing certificates that are be used by the package source. This resource allows clients to detect
+whether a repository signed package has been tampered or has an unexpected signing certificate.
+
+The resource used for fetching this repository signature information is the `RepositorySignatures` resource found in
+the [service index](service-index.md).
+
+## Versioning
+
+The following `@type` value is used:
+
+@type value                | Notes
+-------------------------- | -----
+RepositorySignatures/4.7.0 | The initial release
+
+## Base URL
+
+The entry point URL for the following APIs is the value of the `@id` property associated with the aforementioned
+resource `@type` value. This topic uses the placeholder URL `{@id}`.
+
+Note that unlike other resources, the `{@id}` URL is required to be served over HTTPS.
+
+## HTTP methods
+
+All URLs found in the repository signatures resource support only the HTTP methods `GET` and `HEAD`.
+
+## Repository signatures index
+
+The repository signatures index contains two pieces of information:
+
+1. Whether or not all packages found on the source are repository signed.
+1. The list of certificates used by the package source to sign packages.
+
+In most cases, the list of certificates will only ever be appended to. New certificates would be added to the list when
+the previous signing certificate has expired and the package source needs to start using a new signing certificate. If
+a certificate is removed from the list, that means that all packages signed with the removed signing certificate should
+not longer be considered valid by the client.
+
+In the case of certificate revocation (e.g. key compromise), the package source is expected to resign all packages
+signed by the affected certificate. Additionally, the package source should remove the affected certificate from the
+signing certificate list.
+
+The following request fetches the repository signatures index.
+
+    GET {@id}
+
+The repository signature index is a JSON document that contains an object with the following properties:
+
+Name                | Type             | Required
+------------------- | ---------------- | --------
+allRepositorySigned | boolean          | yes
+signingCertificates | array of objects | yes
+
+The `allRepositorySigned` boolean is set to false if the package source has some packages that have no repository
+signature. If the boolean is set to true, all packages available on the source must have a repository
+signature produced by one of the signing certificates mentioned in `signingCertificates`.
+
+There may be zero or more signing certificates in the `signingCertificates` array. Each element in this array is a JSON
```

Comment:
There must be at least one signing certificate if `allRepositorySigned` is true?

Reply:
Spoke offline. This is a possible state.

Reply:
Fixed.